### PR TITLE
docs(tests): add instructions to run with cmake

### PR DESCRIPTION
**Pull request by kylo252**
The provided Makefile should be a thin wrapper for triggering a few CMake commands, and while it's generally helpful, it's more beneficial to show the actual command to run these tests without bundling GNU Make as a requirement.

A few points worth mentioning regarding `make functionaltest`

- it will trigger CMake on `.deps` every time you run it, which is a good idea in theory, but it's something that should be left to CMake to decide on.
- it's not ideal for distributions, or any user that want to run tests but without the bundled deps, you can turn most of bundled deps off, but that requires setting Make variables which are then sent to CMake for deps you didn't even need in the first place.

### CODE REVIEW

**Comment by gpanders** [comment]
Maybe this discussion already happened elsewhere, but why is 

```
cmake --build build --target functionaltest
```

preferable to just `make functionaltest`?

By the way it’s a good idea to put the explanation/justification for changes in the commit and PR description.

**Comment by kylo252** [comment]
> Maybe this discussion already happened elsewhere, but why is
> 
> ```
> cmake --build build --target functionaltest
> ```
> 
> preferable to just `make functionaltest`?

it's more of an indication that Make is not requirement for running those tests. it also helps avoid trying to explain that there's a Makefile which is a wrapper around CMake which can produce either ninja or makefile scripts.

also `make functionaltest` will actually trigger cmake on `.deps` every time you run it, which is a good idea in theory, but it's something that should be left to cmake to decide.

> By the way it’s a good idea to put the explanation/justification for changes in the commit and PR description.

sorry, it's a continuation of a discussion I've had a few time with @justinmk 😅 

I'll try to prepare a summary.

File: test/README.md [hunk 1]
```diff
@@ -54,69 +54,16 @@ Layout
...
-    make oldtest TEST_FILE=test_syntax.vim
-
-or:
-
-    make src/nvim/testdir/test_syntax.vim
-
-- Specify only the test file name, not the full path.
-
-
-Debugging tests
```
Comment (justinmk) [inline test/README.md:91]: this was here for a reason: it is a more common task, so it's near the top

File: test/README.md [hunk 2]
```diff
@@ -54,69 +54,16 @@ Layout
...
-    local screen = Screen.new()
-    screen:attach()
-    ```
-
-  Afterwards, put `screen:snapshot_util()` at any position in your test. See the
-  comment at the top of `test/functional/ui/screen.lua` for more.
-
-Filtering Tests
----------------
+```console
```
Comment (justinmk) [inline test/README.md:59]: do not "update" all the shell code to be highlighted. It was in indented form because that's more readable as a text file. Highlighting is not very useful for these small snippets, anyway.

File: test/README.md [hunk 3]
```diff
@@ -54,69 +54,16 @@ Layout
...
+cmake --build build --target unittest
+cmake --build build --target functionaltest
+```
 
 ### Filter by name
 
-Another filter method is by setting a pattern of test name to `TEST_FILTER` or `TEST_FILTER_OUT`.
+Another filter method is by setting a pattern of test name as environment variables `TEST_FILTER` or `TEST_FILTER_OUT`.
```
Comment (justinmk) [inline test/README.md:66]: if it's not clear there are env vars then prefix them with `$`. I don't know why people love to write and read "environment variables", which is extremely verbose.

File: test/README.md [hunk 4]
```diff
@@ -127,27 +74,29 @@ it('bar api',function()
 end)
 ```
 
-To run only test with filter name:
+Run only test with filter name
```
Comment (justinmk) [inline test/README.md:77]: why did you remove the colons in these passages?

File: test/README.md [hunk 5]
```diff
@@ -127,27 +74,29 @@ it('bar api',function()
...
-    TEST_FILE=test/functional/foo.lua make functionaltest
+```console
+# unittest
+TEST_FILE=test/unit/foo.lua cmake --build build --target unittest
 
-To *repeat* a test:
-
-    BUSTED_ARGS="--repeat=100 --no-keep-going" TEST_FILE=test/functional/foo_spec.lua make functionaltest
+# functional
+TEST_FILE=test/functional/foo.lua cmake --build build --target functionaltest
```
Comment (justinmk) [inline test/README.md:98]: `foo_spec` was an extra hint. Why remove the `_spec` part?

**Comment by justinmk** [comment]
Maybe I didn't edit https://github.com/neovim/neovim/pull/17090#issuecomment-1133877963 quickly enough. I don't see many useful changes here.

CMake is the source of truth. But the README.md is also meant to make it as easy as possible for people to get started. 

Given that the only real change in this PR is to change `make` to `cmake --build build` over and over, I'm not going to spend more time on this.
